### PR TITLE
Add Control Tower admin role

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR creates a new Control Tower admin role in the Master account and makes it assumable from the Users account.

## 💭 Motivation and context ##

In order to more effectively use [`cisagov/provision-aws-account`](https://github.com/cisagov/provision-aws-account), we require an IAM role that has all of the permissions necessary to provision AWS accounts via [Control Tower](https://aws.amazon.com/controltower/).  

Up until now, new accounts were being provisioned using [`cisagov/provision-aws-account`](https://github.com/cisagov/provision-aws-account) and temporary admin credentials obtained from [AWS SSO](https://docs.aws.amazon.com/singlesignon/).  The problem is that those credentials are only good for an hour, which is not a very long time considering it can take up to 30 minutes to provision a single new account using Control Tower.  We will need to provision a large number of AWS accounts to support our needs, hence the 12-hour maximum session duration on the Control Tower Admin role defined in this PR.

## 🧪 Testing ##

First, I imported our existing Control Tower Account Factory portfolio into our Terraform state.  Then I successfully applied the Terraform in our Master account.  After that, I confirmed that I was able to use the new Control Tower Admin role to provision multiple new AWS accounts using [`cisagov/provision-aws-account`](https://github.com/cisagov/provision-aws-account).

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
Huzzah!  Adventure awaits!

**Review by jsf9k — APPROVED** [review]
_Approval intensifies_

File: master/README.md [hunk 1]
```diff
@@ -78,17 +78,27 @@ future changes by simply running `terraform apply
...
 | [aws_iam_policy_document.assume_role_doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) | data source |
+| [aws_iam_policy_document.assume_role_users_control_tower_doc](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) | data source |
 | [aws_organizations_organization.cool](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/organizations_organization) | data source |
 
 ## Inputs ##
 
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
 | aws\_region | The AWS region where the non-global resources for the Master account are to be provisioned (e.g. "us-east-1"). | `string` | `"us-east-1"` | no |
+| controltoweradmin\_role\_description | The description to associate with IAM role that allows all necessary permissions to provision AWS accounts via Control Tower in the Master account. | `string` | `"Allows all necessary permissions to provision AWS accounts via Control Tower in the Master account."` | no |
```
Comment (mcdonnnj) [inline master/README.md]:
```suggestion
| controltoweradmin\_role\_description | The description to associate with the IAM role that allows all necessary permissions to provision AWS accounts via Control Tower in the Master account. | `string` | `"Allows all necessary permissions to provision AWS accounts via Control Tower in the Master account."` | no |
```

File: master/variables.tf [hunk 2]
```diff
@@ -10,6 +10,18 @@ variable "aws_region" {
   default     = "us-east-1"
 }
 
+variable "controltoweradmin_role_description" {
+  type        = string
+  description = "The description to associate with IAM role that allows all necessary permissions to provision AWS accounts via Control Tower in the Master account."
```
Comment (mcdonnnj) [inline master/variables.tf]:
```suggestion
  description = "The description to associate with the IAM role that allows all necessary permissions to provision AWS accounts via Control Tower in the Master account."
```

**Review by mcdonnnj — APPROVED** [review]
Strong work 💪💪💪 Please see my pair of connected changes to fix a missing word, but otherwise LGTM ✔

**Pull request merged**